src/Build/Evaluation/Evaluator.cs

```diff
@@ -223,12 +224,8 @@ internal class Evaluator<P, I, M, D>
                 buildEventContext,
                 string.IsNullOrEmpty(projectRootElement.ProjectFileLocation.File) ? "(null)" : projectRootElement.ProjectFileLocation.File);
 
-            // If someone sets the 'MsBuildLogPropertyTracking' environment variable to a non-zero value, wrap property accesses for event reporting.
-            if (Traits.Instance.LogPropertyTracking > 0)
-            {
-                // Wrap the IEvaluatorData<> object passed in.
-                data = new PropertyTrackingEvaluatorDataWrapper<P, I, M, D>(data, _evaluationLoggingContext, Traits.Instance.LogPropertyTracking);
-            }
+            // Wrap the IEvaluatorData<> object passed in.
+            data = new PropertyTrackingEvaluatorDataWrapper<P, I, M, D>(data, _evaluationLoggingContext, Traits.Instance.LogPropertyTracking);
```

Comment:
Is the unconditional wrapping expected to have negligible perf impact?

Reply:
Let's test with PerfStart

Reply:
PerfStar is currently down with unclear timelines. So I run a manual perf tests- numbers included here:

https://github.com/dotnet/msbuild/pull/10125

tl;dr;: Basic measurement did not reveal measurable difference